model/DataQuery.php

```diff
@@ -661,6 +664,15 @@ public function applyRelation($relation) {
 					$this->query->addLeftJoin($componentBaseClass,
 						"\"$relationTable\".\"$componentField\" = \"$componentBaseClass\".\"ID\"");
 				}
+				else {
+					$alias = uniqid($componentBaseClass);
+					$this->query->addLeftJoin(
+						$componentBaseClass,
+						"\"$relationTable\".\"$componentField\" = \"$alias\".\"ID\"",
+						$alias
+					);
+					$this->lastAlias = $alias;
+				}
```

Comment:
We need to distinguish between already existing joins and whether or not we need a new join because we are joining to a new distinct relationship...

Hard to do this as we need overall context of the entire filter for that :/